### DISCUSSION TITLE
Cannot Import dbatools To Use With PowerShell Script Within Docker Container

### DISCUSSION THREAD

**Discussion by thall90** [dataplat/dbatools#8692] [Q&A]
I've created a script that uses `dbatools` to backup a remote database and restore it to another remote database. This script works well on my local machine, however when attempting to run said script within a Docker container built from an image based on the Amazon Linux 2 image, I consistently receive the following error on the line of the script that imports the `dbatools` module: 

`hostname: The term 'hostname' is not recognized as a name of a cmdlet, function, script file, or executable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.`

I've searched across these discussions and the issues list, but wasn't able to find any similar issues aside from [this one](https://github.com/dataplat/dbatools/discussions/8643), which is not the same as the issue that I'm experiencing. I posted here first, rather than creating an issue, because I feel like this is more likely related to a lack of knowledge on my part, rather than a bug. If that's not the case, however, please let me know and I'll create an issue instead. 

I've attached a ZIP file including a trimmed-down version of the Dockerfile and PowerShell script (shown below) that can be used to reproduce the issue. There are also `Build` and `Run` PowerShell scripts included in the ZIP that can be used to create the image and start the container.

Thank you in advance for any assistance!

Test.ps1 :
```powershell
$ErrorActionPreference = "Stop"

Import-Module dbatools # script fails here
```

Dockerfile:
```dockerfile
# Create base layer to install tools which installer layers will use, but not to be used in final runner
FROM amazonlinux:2 as zipper
RUN yum update -y 
RUN yum install -y unixODBC.x86_64 \
  && yum install -y unixODBC-devel.x86_64 \
  && yum install -y util-linux \
  && yum install -y unzip \
  && yum install -y tar \
  && yum install -y gzip \
  && yum install -y gettext \
  && yum install -y jq \
  && yum install -y sudo

# Base layer to copy installation artifacts, confirm tool version information, etc
FROM zipper as base

COPY ./Test.ps1 /Test.ps1

# Download Powershell package
RUN curl https://packages.microsoft.com/config/rhel/8/prod.repo | sudo tee /etc/yum.repos.d/microsoft.repo

# Install PowerShell
RUN sudo yum install -y powershell
RUN pwsh -Version

# Install DBA Tools for PowerShell
RUN pwsh -c "Set-PSRepository -Name PSGallery -InstallationPolicy Trusted"
RUN pwsh -c "Install-Module dbatools -Scope AllUsers -Force" # I've tried several scopes / no scopes here, with no difference in result

ENTRYPOINT [ "pwsh", "Test.ps1" ]
```

[DbaToolsIssueRepro.zip](https://github.com/dataplat/dbatools/files/10263048/DbaToolsIssueRepro.zip)

**Comment by potatoqualitee** [comment]
this should be an easy fix, thanks for the report. it'll be fixed in dbatools 2.0

**Comment by potatoqualitee** [comment]
Hello, can you please run the the following command:
```
Install-Module dbatools -AllowPrerelease -Scope CurrentUser
```

then restart powershell, and try again. 

this is prerelease2 of dbatools 2.0, which contains a ton of fixes
